rocketpy/motors/tank.py

```diff
@@ -410,6 +409,73 @@ def inertia(self):
         """
         return self.liquid_inertia + self.gas_inertia
 
+    def _check_volume_bounds(self):
+        """Checks if the tank is overfilled or underfilled. Raises a ValueError
+        if the tank is overfilled or underfilled.
+        """
+        if (self.fluid_volume > self.geometry.total_volume + 1e-6).any():
+            raise ValueError(
+                f"The tank '{self.name}' is overfilled. The fluid volume is "
+                + "greater than the total volume of the tank.\n\t\t"
+                + "Try increasing the tank height and check out the fluid density"
+                + " values.\n\t\t"
+                + f"The fluid volume is {np.max(self.fluid_volume.y_array):.3f} m³ at "
+                + f"{self.fluid_volume.x_array[np.argmax(self.fluid_volume.y_array)]:.3f} s."
+            )
+        elif (self.fluid_volume < -1e-6).any():
+            raise ValueError(
+                f"The tank '{self.name}' is underfilled. The fluid volume is "
+                + "negative.\n\t\t"
+                + "Try increasing input fluid quantities and check out the fluid density"
+                + " values.\n\t\t"
+                + f"The fluid volume is {np.min(self.fluid_volume.y_array):.3f} m³ at "
+                + f"{self.fluid_volume.x_array[np.argmin(self.fluid_volume.y_array)]:.3f} s."
+            )
+
+    def _check_height_bounds(self):
+        """Checks if the tank is overfilled or underfilled. Raises a ValueError
+        if the tank is overfilled or underfilled.
+        """
+        top_tolerance = self.geometry.top + 1e-4
+        bottom_tolerance = self.geometry.bottom - 1e-4
+
```

Comment:
Can the `geometry.top` or the `geometry.bottom` somehow be equal to zero?

If so, would the bottom_tolerance allow negative values of liquid_height? Could this break something further on in a simulation?

Reply:
The `liquid_height` value is computed relative to the `Tank` geometry reference so it could be negative without any input error.

If you meant that `geometry.top` and `geometry.bottom` could both be zero at the same time, then the tank geometry would degenerate and I don't know precisely its consequences.